server/re/src/icatAdminMS.cpp

```diff
@@ -633,82 +633,6 @@ int msiRenameLocalZoneCollection(msParam_t* _new_zone_name, ruleExecInfo_t* _rei
     return SYS_SERVICE_ROLE_NOT_SUPPORTED;
 }
 
-/**
- * \fn msiAclPolicy(msParam_t *msParam, ruleExecInfo_t *)
- *
- * \brief   When called (e.g. from acAclPolicy) and with "STRICT" as the
- *    argument, this will set the ACL policy (for GeneralQuery) to be
- *    extended (most strict).
- *
- * \module core
- *
- * \since pre-2.1
- *
- *
- * \note Should not be used outside of the rules defined in core.re.
- * Once set STRICT, strict mode remains in force (users can't call it in
- * another rule to change the mode back to non-strict).
- * See core.re.
- *
- * \usage See clients/icommands/test/rules/
- *
- * \param[in] msParam - a msParam of type STR_MS_T - can have value 'STRICT'
- * \param[in,out] - The RuleExecInfo structure that is automatically
- *    handled by the rule engine. The user does not include rei as a
- *    parameter in the rule invocation.
- *
- * \DolVarDependence
- * \DolVarModified
- * \iCatAttrDependence
- * \iCatAttrModified
- * \sideeffect none
- *
- * \return integer
- * \retval status
- * \pre none
- * \post none
- * \sa none
- **/
-int
-msiAclPolicy( msParam_t* msParam, ruleExecInfo_t* ) {
-    char *inputArg;
-
-    std::string svc_role;
-    irods::error ret = get_catalog_service_role(svc_role);
-    if(!ret.ok()) {
-        irods::log(PASS(ret));
-        return ret.code();
-    }
-
-    std::string strict = "off";
-    inputArg = ( char * ) msParam->inOutStruct;
-    if ( inputArg != NULL ) {
-        if ( strncmp( inputArg, "STRICT", 6 ) == 0 ) {
-            if( irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role ) {
-                chlGenQueryAccessControlSetup( NULL, NULL, NULL, 0, 2 );
-                strict = "on";
-            }
-        }
-    }
-    else {
-        if( irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role ) {
-            chlGenQueryAccessControlSetup( NULL, NULL, NULL, 0, 0 );
-        }
-    }
-
-    // set a strict acl prop
-    try {
-        irods::set_server_property<std::string>(irods::STRICT_ACL_KW, strict);
-    } catch ( irods::exception& e ) {
-        irods::log(e);
-        return e.code();
-    }
```

Comment:
You're going to need to port this into agent_main.cpp (i.e. the agent factory).

Notice the check on L686.

Reply:
Port it, but without the options to change it from STRICT mode.

Reply:
Spoiler for a commit I haven't pushed yet (going to right after). I put the following from the function you highlighted at the old call site of the static PEP:
```c++
if (irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role) {
        chlGenQueryAccessControlSetup(NULL, NULL, NULL, 0, 2);
}
```

Do we set this up in agent main?

I'm still seeing more calls to `chlGenQueryAccessControlSetup(...)` in the static PEP call-site: https://github.com/irods/irods/blob/a73a90cf0ba46c86f7cb6e9c5eae7e132faeb147/server/api/src/rsGenQuery.cpp#L840-L855

Reply:
> I put the following from the function you highlighted at the old call site of the static PEP.

Old call site of the static PEP? Got a link to a source file and line number?

> Do we set this up in agent main?

`rsGenQuery` is invoked quite early in the process and is the first function which triggers the PEP. I suspect that's why you're seeing multiple invocations of `chlGenQueryAccessControlSetup`.

Reply:
> > I put the following from the function you highlighted at the old call site of the static PEP.
> 
> Old call site of the static PEP? Got a link to a source file and line number?

Here it is: https://github.com/irods/irods/blob/a73a90cf0ba46c86f7cb6e9c5eae7e132faeb147/server/api/src/rsGenQuery.cpp#L808-L834

> > Do we set this up in agent main?
> 
> `rsGenQuery` is invoked quite early in the process and is the first function which triggers the PEP. I suspect that's why you're seeing multiple invocations of `chlGenQueryAccessControlSetup`.

Yeah, kinda figured. Not sure what all the magic numbers at the end mean, so that's why I put the call at the old call site.

Reply:
I think all of that can be removed in favor of setting the necessary values in agent_main.cpp. There's no reason to have GenQuery1 be responsible for that anymore.

Reply:
Moving it to the agent factory's init logic means it will be invoked one time too.

Reply:
Basically, somewhere before this point.

https://github.com/irods/irods/blob/a73a90cf0ba46c86f7cb6e9c5eae7e132faeb147/server/main_server/src/agent_main.cpp#L300

Reply:
Can the `irods::KW_CFG_SERVICE_ROLE_PROVIDER == svc_role` check be done at that point?

Reply:
It should be possible to do the check there.

Reply:
Will try and then take out of draft for building and testing.

Reply:
Done.